spec/unit/iiif/v3/presentation/resource_spec.rb

```diff
@@ -1,19 +1,174 @@
-    it_behaves_like 'it has the appropriate methods for numeric-only keys v3'
+  describe '#validate' do
+    it 'raises an IllegalValueError if id is not URI' do
+      subject['id'] = 'foo'
+      subject['type'] = 'image/jpeg'
+      exp_err_msg = "id must be an http(s) URI for #{described_class}"
+      expect { subject.validate }.to raise_error(IIIF::V3::Presentation::IllegalValueError, exp_err_msg)
+    end
+    it 'raises an IllegalValueError if id is not http' do
+      subject['id'] = 'ftp://www.example.org'
+      subject['type'] = 'image/jpeg'
+      exp_err_msg = "id must be an http(s) URI for #{described_class}"
+      expect { subject.validate }.to raise_error(IIIF::V3::Presentation::IllegalValueError, exp_err_msg)
+    end
   end
 
-  describe "#{described_class}.define_methods_for_string_only_keys" do
-    it_behaves_like 'it has the appropriate methods for string-only keys v3'
+  describe 'realistic examples' do
+    describe 'non-image examples from http://prezi3.iiif.io/api/presentation/3.0' do
+      describe 'audio' do
+        let(:file_id) { 'http://example.org/iiif/book1/res/music.mp3' }
+        let(:file_type) { 'dctypes:Sound' }
+        let(:file_mimetype) { 'audio/mpeg' }
+        let(:resource_object) { IIIF::V3::Presentation::Resource.new({
+            'id' => file_id,
+            'type' => file_type,
+            'format' => file_mimetype
+        })}
+        it 'validates' do
+          expect{resource_object.validate}.not_to raise_error
+        end
+        it 'has expected required values' do
+          expect(resource_object.id).to eq file_id
+        end
+        it 'has expected other values' do
+          expect(resource_object.type).to eq file_type
+          expect(resource_object.format).to eq file_mimetype
+          expect(resource_object.service).to eq []
+        end
+      end
+      describe 'text' do
+        let(:file_id) { 'http://example.org/iiif/book1/res/tei-text-p1.xml' }
+        let(:file_type) { 'dctypes:Text' }
+        let(:file_mimetype) { 'application/tei+xml' }
+        let(:resource_object) { IIIF::V3::Presentation::Resource.new({
+            'id' => file_id,
+            'type' => file_type,
+            'format' => file_mimetype
+        })}
+        it 'validates' do
+          expect{resource_object.validate}.not_to raise_error
+        end
+        it 'has expected required values' do
+          expect(resource_object.id).to eq file_id
+        end
+        it 'has expected other values' do
+          expect(resource_object.type).to eq file_type
+          expect(resource_object.format).to eq file_mimetype
+          expect(resource_object.service).to eq []
+        end
+      end
+
+    end
+    describe 'stanford' do
+      describe 'non-image resource per purl code' do
+        let(:file_id) { 'https://example.org/file/abc666/ocr.txt' }
+        let(:file_type) { 'Document' }
+        let(:file_mimetype) { 'text/plain' }
+        let(:resource_object) {
```

Comment:
oh yeah - I've seen that too.  At this point, I think I'm inclined to just let it ride.